src/main/java/com/aws/greengrass/smbridge/clients/SMClient.java

```diff
@@ -0,0 +1,127 @@
+import com.aws.greengrass.componentmanager.KernelConfigResolver;
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+import com.aws.greengrass.smbridge.StreamMessage;
+import com.aws.greengrass.smbridge.StreamDefinition;
+import com.amazonaws.greengrass.streammanager.client.StreamManagerClient;
+
+import javax.inject.Inject;
+import java.util.List;
+
+public class SMClient {
+    private static final Logger LOGGER = LogManager.getLogger(SMClient.class);
+
+    private StreamManagerClient smClient;
+    private MessageStreamDefinition defaultStream;
+    /**
+     * Ctr for SMClient.
+     *
+     * @param topics             topics passed in by Nucleus
+     * @throws SMClientException if unable to create SM Client
+     */
+    @Inject
+    public SMClient(Topics topics) throws SMClientException {
+        this(topics, null);
+        // TODO: Handle the case when serverUri is modified
+        try {
+            this.smClient = StreamManagerClientFactory.standard().build();
+        } catch (StreamManagerException e) {
+            throw new SMClientException("Unable to create a SM client", e);
+        }
+    }
+    protected SMClient(Topics topics, StreamManagerClient streamManagerClient){
+        this.smClient = streamManagerClient;
+        // TODO: Configurable default stream
+        this.defaultStream = new MessageStreamDefinition();
+        this.defaultStream.setName("mqttToStreamDefaultStreamName");
+        this.defaultStream.setTimeToLiveMillis(9223372036854L);
+        this.defaultStream.setStrategyOnFull(StrategyOnFull.RejectNewData);
+    }
+
+    // TODO: Connect to stream manager on custom port
+    //  https://docs.aws.amazon.com/greengrass/v2/developerguide/use-stream-manager-in-custom-components.html
+    public void start() throws SMClientException {
+        try {
+            updateOrCreateStream(defaultStream);
+        } catch (StreamManagerException e) {
+            LOGGER.atError().log("Encountered StreamManagerException while starting SM Client");
+            throw new SMClientException("Unable to start SMClient");
+        }
+    }
+
+    public void publishOnDefaultStream(byte[] payload) throws SMClientException{
+        // TODO: Configurable default stream name
+        String defaultStreamName = "mqttToStreamDefaultStreamName";
+        publish(new StreamMessage(defaultStreamName, payload));
+    }
+
+    public void publish(StreamMessage message) throws SMClientException{
+        try {
+            if (!checkStreamExists(message.getStream())) {
+                // TODO: Configurable default stream
+                MessageStreamDefinition newStream = new MessageStreamDefinition();
+                newStream.setName(message.getStream());
+                newStream.setTimeToLiveMillis(9223372036854L);
+                newStream.setStrategyOnFull(StrategyOnFull.RejectNewData);
```

Comment:
This logic is duplicated. Can be moved a single place.

Reply:
See ibid